server/controllers/eventController.js

```diff
@@ -0,0 +1,31 @@
+const Club = require("../models/clubModel");
+const Event = require("../models/clubEventModel");
+
+exports.createEvent = async (req, res) => {
+    try {
+        const { title, description, date, location, clubName } = req.body;
+
+        // Check if the club exists
+        const club = await Club.findOne({ name: clubName });
```

Comment:
Aren't club admins the only ones who can create an event for a club? Get session user, then get their role. If it's not admin (including if they aren't a user, i.e signed as guest), throw an authoriztion error. Similar to club editing endpoint